### PR TITLE
Fix/issue/38 1

### PR DESCRIPTION
**Pull request by Yamaguchi**
This is the first pull request for #38 .
This PR will add test cases for process_blocksig function.

### CODE REVIEW

File: src/signer_node/message_processor/process_blocksig.rs [hunk 1]
```diff
@@ -158,3 +158,380 @@ where
...
+        // then node should
+        //  - call rpc submitblock
+        //  - send message `Completedblock`
+        //  - return Master
+
+        let contents = load_test_vector("./tests/resources/test_vectors.json").unwrap();
+
+        let conman = TestConnectionManager::new();
+        let mut rpc = MockRpc::new();
+        rpc.should_call_submitblock(Ok(()));
```
Comment (rantan) [inline src/signer_node/message_processor/process_blocksig.rs]: You should call `rpc.assert();` after finished using rpc mock for checking there are any remaining stub.
See document in `src/tests/helper/rpc.rs`

File: src/tests/helper/node_state_builder.rs [hunk 2]
```diff
@@ -1,12 +1,85 @@
...
 use crate::signer_node::{BidirectionalSharedSecretMap, NodeState};
 use curv::{FE, GE};
+use std::collections::BTreeMap;
 
 pub trait Builder {
     fn new() -> Self;
     fn build(&self) -> NodeState;
 }
 
+pub struct Master {
```
Comment (rantan) [inline src/tests/helper/node_state_builder.rs]: 👍

File: src/signer_node/message_processor/process_blocksig.rs [hunk 3]
```diff
@@ -158,3 +158,380 @@ where
...
+                assert_eq!(signatures.len(), 2);
+                assert_eq!(round_is_done, true);
+            }
+            _ => {
+                panic!("NodeState should be Master");
+            }
+        }
+    }
+
+    fn load_test_case(
```
[the remark this answers is not in the document]
Comment (Yamaguchi) [reply src/signer_node/message_processor/process_blocksig.rs]: I'll try it. Thx.

File: src/signer_node/message_processor/process_blocksig.rs (hunk 1 above)
Comment (Yamaguchi) [reply src/signer_node/message_processor/process_blocksig.rs]: it is called at line:410

File: src/signer_node/message_processor/process_blocksig.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Yamaguchi) [reply src/signer_node/message_processor/process_blocksig.rs]: bitcoin::PublicKey, bitcoin::PrivateKey, blockdata::Block and SignerID have already implemented Deserialize/Serialize traits, but they serialize its data as array of byte ([u8]) format, not hex string format.
it is difficult to deserialize them from hex string in test_vector json.

But I will apply your idea and use serde_json::from_value to simplify test code.

File: src/signer_node/message_processor/process_blocksig.rs (hunk 1 above)
Comment (rantan) [reply src/signer_node/message_processor/process_blocksig.rs]: Oh sorry, I missed it.

**Pull request merged**